### ISSUE TITLE
Investigate ways to reduce the analysis message matcher overhead

### ISSUE THREAD

**Issue by trink** [mozilla-services/hindsight#135]

**Comment by trink** [comment]
This is a work in progress, experimentation will continue as the schedule allows.

**Comment by cvuillemez** [comment]
+1 :)
I have very long message matchers which are bottlenecks.

**Comment by trink** [comment]
Yeah https://github.com/mozilla-services/lua_sandbox/issues/213 is about all I can squeeze out of a single matcher.

https://github.com/mozilla-services/lua_sandbox/issues/208 may be relevant if the string at the end is unique so you don't need to actually anchor it.  `Type =~ '/unique'` is multiple times faster than `Type =~ '/unique$'`

**Comment by cvuillemez** [comment]
Yeah I tested and it's faster without the trailing "$" .
That's amazing, usually we could think the "$" is faster cause not all string must be parsed, just the end !
